### ISSUE TITLE
read_csv should default to index_col = 0

### ISSUE THREAD

**Issue by itko** [pandas-dev/pandas#24468]
#### Code Sample

```python
df.to_csv(file_path)
df = pd.read_csv(file_path)
```
#### Problem description

Currently, the default CSV writing behaviour is to write the index column.  The default reading behaviour, however, is to assume there is no index column in the file. This is not intuitive when writing and reading files.

The expected behaviour is that a file which is written without any index option it should be able to be read without any index option. However, the default writing and reading behaviour results in an unnamed column.

One fix is to change the default index_col for read_csv, another option is to change the default index boolean for to_csv. The former is probably preferable as it preserves information.

**Comment by chris-b1** [comment]
At this point, both of those defaults are unlikely to be changed.

`pd.DataFrame.from_csv` exists for easier round-tripping, though deprecated, see e.g. #10163

Because it is schema-less, csv is never a particularly safe format for round-tripping, consider using something binary like parquet of HDF5 instead

**Comment by gfyoung** [comment]
> At this point, both of those defaults are unlikely to be changed.

Agreed.  Though might entertain this option more in a super-breaking release like `1.0`.

**Comment by JeroenDelcour** [comment]
This has been rejected before in #12627 with the following reply:

> Has been this way almost since the beginning.
>
> The idea is that .to_csv and .from_csv are inverses
>
> Essentially impossible to change at this point. But to be honest its actually a sensible default. Indexes are more and more important. If you are not using them you should.

However, `.from_csv` has been deprecated in favor of `.read_csv`, which would only be the inverse of `.to_csv` if this default were changed.

I can only interpret the other argument (similar to @chris-b1's reply) to be "because legacy". I personally consider this to be a poor argument in any case. Defaults have been changed before, is there a specific reason this one shouldn't?

> Because it is schema-less, csv is never a particularly safe format for round-tripping, consider using something binary like parquet of HDF5 instead

Agreed. However, while CSV may not be the best data format for round-tripping, in practice this is one of the most common use-cases. Many new users are confused as to why unnamed columns appear in their files seemingly at random. After years of using Pandas, I still regularly forget to set the right arguments to allow round-tripping. In my opinion, the inconsistency of the current defaults only adds unnecessary cognitive load.

I think this is a good issue for v0.25.

**Comment by chris-b1** [comment]
just personally, I'd be more sympathetic to changing the default on `to_csv` to `index=False`, but that has its own set of problems

**Comment by gfyoung** [comment]
> but that has its own set of problems

Agreed.  But the point is well taken that we should pick a suitable (and consistent) default for both to avoid the confusions described above.  That being said, it would be good to get some more opinions on what people generally use in the wild (i.e. with an index, without) before settling on one.

> Defaults have been changed before, is there a specific reason this one shouldn't?

We aren't saying that it shouldn't, but asking for us to do this in `0.25.0` is rushing things IMO.

**Comment by JeroenDelcour** [comment]
> asking for us to do this in `0.25.0` is rushing things IMO.

Sorry, I didn't mean to rush it. I'm not very familiar with the Pandas release cycle. As long as it's not pushed back to `1.0.0` - I don't think it's that breaking.

> just personally, I'd be more sympathetic to changing the default on to_csv to index=False, but that has its own set of problems

I'm leaning towards this, too, if nothing else because it would match what most users I know already do.

**Comment by TomAugspurger** [comment]
Agreed with https://github.com/pandas-dev/pandas/issues/24468#issuecomment-450406181 that this is too large of a change for us at this point. What do you think @itko?

**Comment by itko** [comment]
I tend to agree with @JeroenDelcour. Not only have I seen multiple users get confused by the appearance of unnamed columns, I myself often forget to set the index arguments.

It seems to me we should be prioritizing usability and intuitiveness. Can't we schedule this for release and add a FutureWarning?

**Comment by jbrockmendel** [comment]
changing read_csv behavior affects a zillion users, so is a painful deprecation.  `to_csv` is dramatically less common, so deprecating/changing the behavior there (xref #46583) would be less painful.  But still painful, and still potentially lossy if there is a non-default Index attached.  So I'm skeptical here.

**Comment by pevogam** [comment]
@jbrockmendel So your suggestion is that to have this unintuitive API forever because nothing can be done about it and the project cannot go through a higher enough bump of pain? This then means that any API decisions that didn't generalize well and will cause a bump beyond certain threshold are never to be crossed again and the API should be stratified to this basin of pain tolerance for all eternity.

**Comment by pevogam** [comment]
(I mean this because you are also labeling this as Closing Candidate on top of your comments)

**Comment by JeroenDelcour** [comment]
While changing defaults is mildly inconvenient for existing users, not changing is painful for new users. In the long term, I'd argue streamlining the learning curve is worth the minor inconvenience of a FutureWarning. I'd wager most experienced users set this argument explitictly, anyway.

**Comment by jbrockmendel** [comment]
> [...] because nothing can be done about it and the project cannot go through a higher enough bump of pain?  [...]

Yes, actually.  No one is happy about it, but often times the decision ends up being to keep a sub-optimal API to avoid inconveniencing a larger number of users.

> (I mean this because you are also labeling this as Closing Candidate on top of your comments)

This is my way of trying to draw attention to this from other maintainers.  cc @rhshadrach @mroeschke

**Comment by rhshadrach** [comment]
Related: #12627, #34576, #46583

Changing `read_csv` to default to `index_col=0` would only be beneficial on CSVs that were written by pandas, and even then, only those without a MultiIndex. I would hazard a guess that these are a minority. So not only would it be a lot of churn, it would not be beneficial for the majority.

Having `index=False` in to_csv is silently lossy, and having it depend on the data in the index is complex and may be surprising. I'm fairly strongly opposed here.

While I agree that the behavior isn't perfect, it seems to me all proposed alternatives are strictly less optimal.

@mroeschke - thoughts here?

**Comment by JeroenDelcour** [comment]
> Having index=False in to_csv is silently lossy

That's a good point I hadn't considered. Better to include superfluous data than silently skip data.

Looking at the other output functions, `index=True` appears to be the standard. `to_csv` being the odd one out could catch users unaware, making accidental data loss more likely.

I'm starting to lean towards keeping it `index=True`.
